src/NeinLinq/InjectLambdaSignature.cs

```diff
@@ -152,7 +152,8 @@ private static Exception FailFactory(Type target, string method, string error)
 
             result = candidate;
         }
-
-        return result;
+        return result is not null || target.BaseType is not { } baseTarget
```

Comment:
Inverting this condition should make easier, a bit

`return result is null && target.BaseType is var baseTarget ...`

Reply:
You are right. I updated this part. It would be possible to switch the recursion with a while loop. 
But I think this would not help to make it more clear.

Reply:
Nah, the recursion is fine 👍